lightning/src/ln/channel.rs

```diff
@@ -686,6 +686,43 @@ impl UnfundedChannelContext {
 	}
 }
 
+/// Info about a transaction and its confirmation status, used mainly for funding transactions.
+pub(super) struct TransactionConfirmation {
+	/// The transaction, or None.
+	transaction: Option<Transaction>,
+	/// The hash of the block in which the transaction was included, or None.
+	confirmed_in: Option<BlockHash>,
+	/// The height of the block in which the transaction was included, or 0.
+	confirmation_height: u32,
+}
+
+impl TransactionConfirmation {
+	/// Construct with empty values
+	fn default() -> Self {
+		Self {
+			transaction: None,
+			confirmed_in: None,
+			confirmation_height: 0,
+		}
+	}
+
+	/// Get the confirmation depth: height relative to the given current height.
+	/// Also returns a flag indicating the special case when the confirmation is in the 'future'.
+	/// If there is no confirmation height (it was not confirmed, or confirmed and reorged): (0, false)
+	/// If the confirmation height is in the 'future' (e.g. due to a reorg): (0, true)
+	/// Otherwise the result is height - confirmation_height + 1, a number always larger than 0.
+	fn confirmation_depth(&self, current_height: u32) -> (u32, bool) {
```

Comment:
Instead of using a `bool`, I'd recommend encoding the possibilities with an `enum` (e.g., using an `Unconfirmed` variant or such). Otherwise, there is a fourth possibility that is undefined. Also, it would make call sites more understandable than relying on correctly interpreting a `bool`. For instance, you could easily confuse `true` with meaning it is confirmed rather than it will be confirmed.

Reply:
Introduced enum for relative confirmation height.